crates/volta-core/src/run/binary.rs

```diff
@@ -20,6 +20,17 @@ pub(super) fn command(exe: &OsStr, args: &[OsString], session: &mut Session) ->
     if let Some(project) = session.project()? {
         // Check if the executable is a direct dependency
         if project.has_direct_bin(exe)? {
+            if project
+                .platform()
+                .map_or(false, |platform| platform.yarn.is_some())
+                && project.is_plug_n_play()
```

Comment:
Thoughts on rolling the check for `platform.yarn` into the `is_plug_n_play` method? Perhaps renaming to `is_yarn_pnp` or similar?

Reply:
Good idea! I combined that into `is_yarn_pnp`